client/src/main/java/io/pravega/client/stream/impl/EventPointerImpl.java

```diff
@@ -52,4 +61,62 @@ int getEventLength() {
     public EventPointerInternal asImpl() {
         return this;
     }
+    
+    public static EventPointer fromString(String eventPointer) {
+        int i = eventPointer.lastIndexOf(":");
+        Preconditions.checkArgument(i > 0, "Invalid event pointer: %s", eventPointer);
+        String[] offset = eventPointer.substring(i + 1).split("-");
+        Preconditions.checkArgument(offset.length == 2, "Invalid event pointer: %s", eventPointer);
+        return new EventPointerImpl(Segment.fromScopedName(eventPointer.substring(0, i)), Long.parseLong(offset[0]),
+                                    Integer.parseInt(offset[1]));
+    }
+    
+    @Override
+    public String toString() {
+        return segment.getScopedName() + ":" + eventStartOffset + "-" + eventLength;
+    }
+
+    private static class EventPointerBuilder implements ObjectBuilder<EventPointerImpl> {
+    }
+    
+    public static class EventPointerSerializer extends VersionedSerializer.WithBuilder<EventPointerImpl, EventPointerBuilder> {
+        @Override
+        protected EventPointerBuilder newBuilder() {
+            return builder();
+        }
+
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, EventPointerBuilder builder) throws IOException {
+            builder.segment(Segment.fromScopedName(revisionDataInput.readUTF()));
+            builder.eventStartOffset(revisionDataInput.readCompactLong());
```

Comment:
Make sure you understand the constraints of compactInt and compactLong: non-negative values up to 2^30 (int) or 2^62 (long).

Reply:
yes, I am assuming we don't have more than 2^30 segments in a stream and 2^62 bytes in a segment.

Reply:
ok